src/AzureStore.cs

```diff
@@ -219,19 +211,19 @@ public async Task<IAzureReadBlobRef> CommitTemporaryBlob(string name, Cancellati
 
             var hash = new Hash(md5.Hash);
 
-            var final = Persistent.GetBlockBlobReference(AzureBlobName(Realm, hash));
+            var final = Persistent.GetBlobClient(AzureBlobName(Realm, hash));
 
             try
             {
-                var exists = await AzureRetry.OrFalse(() => final.ExistsAsync(null, null, cancel))
+                var exists = await AzureRetry.OrFalse(async () => await final.ExistsAsync(cancellationToken: cancel))
                     .ConfigureAwait(false);
 
                 if (!exists)
                 {
                     await CopyToPersistent(temporary, final, cancel).ConfigureAwait(false);
                 }
 
-                _onCommit?.Invoke(sw.Elapsed, Realm, hash, final.Properties.Length, exists);
+                _onCommit?.Invoke(sw.Elapsed, Realm, hash, final.GetProperties()?.Value?.ContentLength ?? 0, exists);
```

Comment:
If `GetProperties()?.Value?.ContentLength` is null, it would be safer to throw an exception than to continue with the incorrect value 0.